### ISSUE TITLE
Question about return status

### ISSUE THREAD

**Issue by yangcen6** [allegro/allegro-api#10150]
1. Is the **finished** status always available in **/order/customer-returns/{customerReturnId}**?
![20241023-160727](https://github.com/user-attachments/assets/a0cac6a0-b448-414e-930c-80956c800b38)
2. Why do we only receive **COMMISSION_REFUNDED** for these orders? 408f6ba0-89bd-11ef-95de-87deba44aa82，489a0180-8ade-11ef-b890-a7a79bb8f571，0d7ccbf0-8ade-11ef-9806-0b2fec43d54a

**Comment by Lukasz-Zurek** [comment]
1. We can only return one specific status, so it depends on situation. If the payment has been refunded and return process is already finished - we return FINISHED, but if you decided to create commission refund application - the status will be changed to COMMISSION_REFUND_CLAIMED and then to COMMISSION_REFUNDED if the sales commission was refunded.
2. Without any detailed context, I am not able to answer this question. If you have a specific case where we return an incorrect status - please give me an example.

**Comment by yangcen6** [comment]
Is it true that only after-sales returns and refunds will the status be changed to "finished"? If the buyer refuses to accept the package and applies for a refund, the status will not be changed to "finished"?

**Comment by Lukasz-Zurek** [comment]
Yes, but please note that if the buyer refuses to accept the package, he won't receive any refund/he won't be able to apply for it.

**Comment by yangcen6** [comment]
But in fact, after the buyer refuses to accept the package, he can still apply for a refund. like this order 489a0180-8ade-11ef-b890-a7a79bb8f571

**Comment by Lukasz-Zurek** [comment]
I have checked customer-return for this specific order and the status is: "COMMISSION_REFUNDED" - which does mean that the seller applied for a commission refund and it has been finished. This is a correct status.

The commission refund is something different than payment return to buyer - where you are right - refusing to accept the package shouldn't cause the customer to not be able to ask for a refund, but in this situation the status will definitely not be finished.

The status "FINISHED" is dedicated for a situation where the payment has been refunded to the buyer. But in this situation the seller can further create a commission refund (to get back his commission for this unrealized sales, so the status will be changed from FINISHED to COMMISSION_REFUND_CLAIMED and then to COMMISSION_REFUNDED.
